lib/chef/provider/template/content.rb

```diff
@@ -40,14 +40,7 @@ def file_for_provider
           visitor = lambda do |obj|
             case obj
             when Hash
-              # If this is an Attribute object, we need to change class otherwise
-              # we get the immutable behavior. This could probably be fixed by
-              # using Hash#transform_values once we only support Ruby 2.4.
-              obj_class = obj.is_a?(Chef::Node::ImmutableMash) ? Mash : obj.class
-              # Avoid mutating hashes in the resource in case we're changing anything.
-              obj.each_with_object(obj_class.new) do |(key, value), memo|
-                memo[key] = visitor.call(value)
-              end
+              obj.transform_values { |value| visitor.call(value) }
```

Comment:
The problem that I can see with this is that when passed an ImmutableMash this will change the return type to a Hash rather than a Mash (probably due to something like `dup` on the ImmutableMash object calling `to_hash` and returning a Hash rather than a Mash, which is a somewhat unfortunate API that we have).

Reply:
Yeah and that's a pretty hard break, i think, this likely breaks any node objects passed into templates where the consumers use symbols as keys rather than strings.  Likely not just a theoretical break.

Reply:
Oh, I see, thanks! I'll take a look and try again later.